### DISCUSSION TITLE
Can anyone help me find the code for the "Hot Corners" / "Show all windows"?

### DISCUSSION THREAD

**Discussion by GalacticSun** [linuxmint/discussions#613] [Development] [answered]
I'm trying to find the code that is responsible for the Hot Corner that shows "Current application windows".  I'd like to make an edit that would show the windows of only the currently active application in the same way that "Show all windows" shows all windows.  For example, when using Firefox and you activate this Hot Corner, it will only show all Firefox windows for easy selection, instead of all the windows you have open.

Related to my post here:
https://github.com/orgs/linuxmint/discussions/410

I just can't seem to find the actual code that runs the "Show all windows" function.  Not even sure what it is written in (hopefully JS).  Can anyone point me in the right direction, please?

**Comment by mtwebster** [comment, marked as answer]
We call the 'show all windows' feature Overview:
https://github.com/linuxmint/Cinnamon/blob/master/js/ui/overview.js

Hot Corners:
https://github.com/linuxmint/Cinnamon/blob/master/js/ui/hotCorner.js

**Reply by GalacticSun** [reply]
Thank you.

**Reply by GalacticSun** [reply]
By the way, @mtwebster, if I wanted to change Overview from showing all windows to just the windows of the current application in focus, where would I start?  Is the code responsible for getting the Overview windows on line 34 ("new St.Widget", etc), and line 46 ("global.overlay_group.add_actor(this._group);")?  Could that, or other code, be adapted to get only one application's set of windows?

**Reply by GalacticSun** [reply]
Any hints, suggestions, at all, @mtwebster ?  Or any other dev?  Just looking for a little bit of help with this.  Thanks.

**Comment by Jopp-gh** [comment]
I'm also interested in "Hot corners", the code is very easy to read. 

I'd like to add multiple modifier keys (shift, Meta, Alt) to add  12 more slots for additional actions. However, to implement a change we have to write an extension, right ?
